web/backend/src/controllers/users.ts

```diff
@@ -34,17 +41,31 @@ usersRouter.post('/add_role', (req, res, next) => {
     }
   }
 
-  addPolicy(req.body.userId, req.body.subject, req.body.permission)
-    .then(() => {
-      res.set(200).send();
-      next();
-    })
-    .catch((e) => {
-      res.status(400).send(`Error while adding to roles: ${e}`);
-    });
-
-  // Call https://search-api.epfl.ch/api/ldap?q=228271, if the answer is
-  // empty then sciper unknown, otherwise add it in userDB
+  if ('userId' in req.body) {
+    try {
+      readSCIPER(req.body.userId);
+      Promise.resolve(addPolicy(req.body.userId, req.body.subject, req.body.permission));
```

Comment:
The `Promise.resolve` doesn't change anything - the promise will not be awaited for. Does the following PR break it?

https://github.com/c4dt/d-voting/pull/159

You can run [Await Playground](https://www.typescriptlang.org/play/?#code/IYZwngdgxgBAZgV2gFwJYHsIwBYFMA2+6AFAJQDeAUDDTFJiOvrgHREDmxARHoel6QDc1WsmwAndAHcYEXDICi4yeO7t06ACYAjMLgHCAvpROhIsRCgxYAtsFQQyVGsnFgYz2jQD03mAHVUMXQEZBhgLAByYCl7ZEicXHFcABoYMVwYJJV4SRsYSN4iBNQQWVwANyS6YAR2bGQWES8ABTzS1mTGfCriIpJSIRFDGuQobBhiXFIPZtp6CG7WbPRVaeEaYxoFpbZ0Th5UAxEYuPKZNvQbDuJicRmAXgA+GBBcZAAVVBtcEOQ7tIARgADKDBhs6AwmKwONwMsljsZKHYHGRhEA) and see that `goodbye` is never called.

Reply:
I wasn't sure about the async in the callback :confused: I'll try with async/await

Reply:
Here is a thread from stackoverflow: https://stackoverflow.com/questions/44813401/passing-in-async-functions-to-node-js-express-js-router

TLDR: 
- express >= 4 handles async callbacks
- express >= 5 handles also errors w/o a try...catch